### PR TITLE
Cleanup streaming app sample

### PR DESCRIPTION
**Pull request by jiria**
**What this PR does / why we need it**:
This PR streamlines the streaming app sample so that both main camera feed and small camera feeds can be dynamically refreshed based on detected cameras

Further work could be done to integrate with Helm and patch up the RBAC to target any namespace

### CODE REVIEW

File: samples/apps/video-streaming-app/streaming.py [hunk 1]
```diff
@@ -19,13 +19,148 @@
...
+    ret = coreV1Api.list_service_for_all_namespaces(watch=False)
+    for svc in ret.items:
+        if svc.metadata.name == configuration_name + "-svc":
+            grpc_ports = list(
+                filter(lambda port: port.name == "grpc", svc.spec.ports))
+            if (len(grpc_ports) == 1):
+                url = "{0}:{1}".format(svc.spec.cluster_ip, grpc_ports[0].port)
+                camera_display.main_camera = CameraFeed(url)
+
     ret = coreV1Api.list_service_for_all_namespaces(watch=False)
```
Comment (bfjelds) [inline samples/apps/video-streaming-app/streaming.py:164]: can we consolidate the 2 for loops here?  or call list_service_for_all_namespaces once?

Comment (jiria) [reply samples/apps/video-streaming-app/streaming.py:164]: Thanks, you are right. They were originally at two different places and I missed refactoring those. Coming up shortly.

**Review by bfjelds — APPROVED** [review]

File: samples/apps/video-streaming-app/streaming.py [hunk 2]
```diff
@@ -19,13 +19,148 @@
...
+        self.queue = queue.Queue(1)
+        self.thread = None
+        self.stop_event = global_stop_event
+
+    def __eq__(self, other):
+        if other is None:
+            return False
+        return self.url == other.url
+
+    def StartHandler(self):
```
[the remark this answers is not in the document]
Comment (jiria) [reply samples/apps/video-streaming-app/streaming.py:36]: Thanks for pointing out, will fix.

**Review by kate-goldenring — APPROVED** [review]

**Comment by jiria** [comment]
What do you propose for the Helm integration? Or leave that for another PR? @kate-goldenring @bfjelds

**Comment by bfjelds** [comment]
> What do you propose for the Helm integration? Or leave that for another PR? @kate-goldenring @bfjelds

helm integration to me is moving the samples yaml into the deployment/helm/templates folder and adding some parameterization.  i think it is fine to be a separate PR.  i also think we should update the ocpua monitoring app with this same type of logic if we can (and then add it to helm)

**Comment by kate-goldenring** [comment]
> > What do you propose for the Helm integration? Or leave that for another PR? @kate-goldenring @bfjelds
> 
> helm integration to me is moving the samples yaml into the deployment/helm/templates folder and adding some parameterization. i think it is fine to be a separate PR. i also think we should update the ocpua monitoring app with this same type of logic if we can (and then add it to helm)

A separate PR to do these two steps makes sense to me. the parameters could be `deploy-streaming-sample` and `deploy-anomaly-sample` or something along those lines

**Comment by kate-goldenring** [comment]
@jiria, can you bump the version twice with `version.sh -u -p`. You want it to be 0.0.1 greater than https://github.com/deislabs/akri/blob/main/version.txt.

**Comment by jiria** [comment]
Version updated, sorry for the delay.

**Review by romoh — APPROVED** [review]

**Review by romoh — APPROVED** [review]
The workflow is failing on the security audit check. I sent a [PR to fix this](https://github.com/deislabs/akri/pull/247).

**Comment by bfjelds** [comment]
it looks like this needs another version bump

**Pull request merged**